go/pkg/client/bytestream_test.go

```diff
@@ -0,0 +1,294 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"google.golang.org/grpc"
+
+	bsgrpc "google.golang.org/genproto/googleapis/bytestream"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
+)
+
+var (
+	logStreams    map[string]*logStream
+	logStreamData = []byte("Hello World! This is large data to send.")
+)
+
+type logStream struct {
+	logStreamID   string
+	logicalOffset int64
+	finalized     bool
+}
+
+func TestWriteBytesWithOffsetSuccess_LogStream(t *testing.T) {
+	tests := []struct {
+		description string
+		ls          *logStream
+		data        []byte
+		numIterate  int
```

Comment:
Nit: I think this could have a more descriptive name that makes it immediately obvious that 1) it is related to the field data and 2) it specifies how many parts that data will be split into. Maybe something like `dataPartsLen` or `dataPartsCount`?

Reply:
Done.